openthread/CMakeLists.txt

```
@@ -6,6 +6,19 @@
 include(${NRFXLIB_DIR}/common.cmake)
 nrfxlib_calculate_lib_path(lib_path)
 
+file (READ "../../modules/lib/openthread/include/openthread/instance.h" INSTANCE_FILE_CONTENT)
+string(REGEX MATCH "#define[ ]*OPENTHREAD_API_VERSION[ ]*\\([0-9]+\\)" MACRO_DEF "${INSTANCE_FILE_CONTENT}")
+string(REGEX MATCH "[0-9]+" OPENTHREAD_SOURCE_API_VERSION "${MACRO_DEF}")
```

No need for the second regex, just make the first regex a matched expression, and extract it using `${CMAKE_MATCH_1}`, like this:
```suggestion
string(REGEX MATCH "#define[ ]*OPENTHREAD_API_VERSION[ ]*\\(([0-9]+)\\)" MACRO_DEF "${INSTANCE_FILE_CONTENT}")
set(OPENTHREAD_SOURCE_API_VERSION ${CMAKE_MATCH_1})
```